app/assets/javascripts/modules/moj.matchVisitors.js

```diff
@@ -0,0 +1,208 @@
+
+      if (unprocessed) {
+        this.showMessage(this.$notAllMessage);
+      } else {
+        this.hideMessage(this.$notAllMessage);
+      }
+    },
+
+    // Get the ages of all selected NOMIS contacts
+    getAges: function() {
+      var self = this,
+        ages = [];
+      $.each(this.getListItems(), function(i, obj) {
+        var data = $(obj).data('visitor'),
+          dob = data.dob.split('-'),
+          age = self.calcAge(new Date(dob[0], dob[1], dob[2]));
+        ages.push(age);
+      });
+      return ages;
+    },
+
+    // Calculate the age from a date of birth
+    calcAge: function(dob) {
+      var ageDifMs = new Date() - dob.getTime(), //Date.now() - dob.getTime(),
+        ageDate = new Date(ageDifMs);
+      return Math.abs(ageDate.getUTCFullYear() - 1970);
+    },
+
+    // Build array of all list items of selected visitors
+    getListItems: function() {
+      var self = this,
+        arr = $('select option:selected').map(function() {
+          var parent = self.findParent(this);
+
+          if (this.value != 0 && !self.isVisitorBanned(parent)) {
+            return parent;
+          }
+        }).get();
+      return arr;
+    },
+
+    // Build an array of chosen visitor values (IDs)
+    getChosenValues: function() {
+      var self = this,
+        arr = $('select').map(function() {
+          if (this.value !== '0') {
+            return this.value
+          }
+        }).get();
+      return arr;
+    },
+
+    // Enable/disable visitor options in the contact list
+    updateSelectLists: function(el) {
+      var self = this,
+        options = this.$el.find('select').not(el).find('option').not(':first');
+
+      $.each(options, function(i, obj) {
+        var contact = $(obj).data('contact');
+
+        if ($.inArray(contact.uid, self.getChosenValues()) !== -1) {
+          $(obj).prop('disabled', 'disabled');
+        } else {
+          $(obj).prop('disabled', null);
+        }
+      });
+    },
+
+    // Find the relating checkbox
+    getNoContactCheckbox: function(el) {
+      var parent = this.findParent(el);
+      return parent.find('input[id*="not_on_list"]');
+    },
+
+    // Enable/disable the 'not on contact list' checkbox
+    setNoContactCheckbox: function(el, disable) {
+      var checkbox = this.getNoContactCheckbox(el);
+      checkbox.prop('disabled', !disable);
+      if (!disable) {
+        checkbox.prop('checked', disable);
+      }
+    },
+
+    // Return true/false
+    isBanned: function(contact) {
+      return contact.banned == 'true';
+    },
+
+    // Auto check the banned checkbox
+    setBanned: function(el, selected) {
+      el.find('input[type="checkbox"][id*="banned"]').prop('checked', selected).trigger('change');
+    },
+
+    // Show the element
+    showMessage: function(el) {
+      el.show().removeClass('visuallyhidden');
+    },
+
+    // Hide the element
+    hideMessage: function(el) {
+      el.hide().addClass('visuallyhidden');
+    }
+
+  };
+}());
```

Comment:
@chrishanes can you rename the function with clearer names so we can remove comments? Basically take the comments and turn them into a function name.
Comments are code smell 😼 !